### ISSUE TITLE
Installation gives Segmentation fault in clean R-3.3.1 install

### ISSUE THREAD

**Issue by jdeligt** [hadley/devtools#1292]
An irrecoverable exception occurred. R is aborting now ...

```
/usr/local/lib/R/bin/INSTALL: line 34:  1549 Done                    echo 'tools:::.install_packages()'
      1550 Segmentation fault: 11  | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
```

installation of package ‘devtools’ had non-zero exit status

**Comment by jdeligt** [comment]
Seems to be in the binary. Full stack trace below;
```
 *** caught segfault ***
address 0x18, cause 'memory not mapped'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
 4: asNamespace(ns)
 5: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
 6: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
 7: asNamespace(ns)
 8: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
 9: loadNamespace(package, lib.loc)
10: doTryCatch(return(expr), name, parentenv, handler)
11: tryCatchOne(expr, names, parentenv, handlers[[1L]])
12: tryCatchList(expr, classes, parentenv, handlers)
13: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        msg <- conditionMessage(e)        sm <- strsplit(msg, "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && identical(getOption("show.error.messages"),         TRUE)) {        cat(msg, file = stderr())        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
14: try({    attr(package, "LibPath") <- which.lib.loc    ns <- loadNamespace(package, lib.loc)    env <- attachNamespace(ns, pos = pos, deps)})
15: library("devtools")
```

**Comment by alexDeCastroAtGit** [comment]
I'm having the same issue: 

 *** caught segfault ***
address 0x18, cause 'memory not mapped'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
 4: asNamespace(ns)
 5: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
 6: loadNamespace(j <- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]])
 7: asNamespace(ns)
 8: namespaceImportFrom(ns, loadNamespace(j <- i[[1L]], c(lib.loc,     .libPaths()), versionCheck = vI[[j]]), i[[2L]], from = package)
 9: loadNamespace(package, lib.loc, keep.source, partial = TRUE)
10: withCallingHandlers(expr, packageStartupMessage = function(c) invokeRestart("muffleMessage"))
11: suppressPackageStartupMessages(loadNamespace(package, lib.loc,     keep.source, partial = TRUE))
12: code2LazyLoadDB(package, lib.loc = lib.loc, keep.source = keep.source,     compress = compress)
13: makeLazyLoading(pkg_name, lib, keep.source = keep.source)
14: doTryCatch(return(expr), name, parentenv, handler)
15: tryCatchOne(expr, names, parentenv, handlers[[1L]])
16: tryCatchList(expr, classes, parentenv, handlers)
17: tryCatch(expr, error = function(e) {    call <- conditionCall(e)    if (!is.null(call)) {        if (identical(call[[1L]], quote(doTryCatch)))             call <- sys.call(-4L)        dcall <- deparse(call)[1L]        prefix <- paste("Error in", dcall, ": ")        LONG <- 75L        msg <- conditionMessage(e)        sm <- strsplit(msg, "\n")[[1L]]        w <- 14L + nchar(dcall, type = "w") + nchar(sm[1L], type = "w")        if (is.na(w))             w <- 14L + nchar(dcall, type = "b") + nchar(sm[1L],                 type = "b")        if (w > LONG)             prefix <- paste0(prefix, "\n  ")    }    else prefix <- "Error : "    msg <- paste0(prefix, conditionMessage(e), "\n")    .Internal(seterrmessage(msg[1L]))    if (!silent && identical(getOption("show.error.messages"),         TRUE)) {        cat(msg, file = stderr())        .Internal(printDeferredWarnings())    }    invisible(structure(msg, class = "try-error", condition = e))})
18: try({    suppressPackageStartupMessages(.getRequiredPackages(quietly = TRUE))    makeLazyLoading(pkg_name, lib, keep.source = keep.source)})
19: do_install_source(pkg_name, instdir, pkg, desc)
20: do_install(pkg)
21: tools:::.install_packages()
An irrecoverable exception occurred. R is aborting now ...
/usr/local/Cellar/r/3.3.1_3/R.framework/Resources/bin/INSTALL: line 34: 94825 Done                    echo 'tools:::.install_packages()'
     94826 Segmentation fault: 11  | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
Warning in install.packages :
  installation of package ‘devtools’ had non-zero exit status

**Comment by hadley** [comment]
Could you please include a minimal reprex? What code are you running to  trigger the segfaults?

**Comment by jdeligt** [comment]
`
R
install.packages("devtools")
`

**Comment by hadley** [comment]
And a `sessionInfo()` please?  (Although I can't imagine that this a bug in devtools)

**Comment by alexDeCastroAtGit** [comment]
Hi Hadley, apologies, I was on leave from work. I think that was an issue with my ROR. In the latest CRAN release I can install devtools fine. 

Thanks,
Alex

**Comment by alexDeCastroAtGit** [comment]
What's "reprex" here?

**Comment by jimhester** [comment]
[Reproducible example] (https://github.com/jennybc/reprex#reproducible-examples)

**Comment by merliseclyde** [comment]
I  had the same issue after upgrading to Mac OS X 10.12.4 and with  R 3.4.0. (same errors)

I removed all /Library/Frameworks/R.frameworks files, reinstalled R and the issues went away (temporarily).

For tracking down another issue I installed R from source via macports, rebuilt packages from source.  However running the binary R from CRAN via the GUI or RStudio the seg faults with loading libraries reappeared.  Reinstalled Xcode, deleted libraries, reinstalled R from CRAN, etc...  but still had seg faults.

Finally looked at .libPath() and realized that both instances of R were looking at the user library before the system library.   So if the packages were linked to different  system libraries from the compiled versions (OS X  Sierra and El Capitan in my case) this could create the seg fault.

Removing libraries in the  user library  ~/user/Library/R/version#  fixed the issue (with installing to different locations)

I believe the issue occured from a  mix of installing packages from install.packages() under both versions without specifying lib  as  the default uses the first entry in  .libPath()  pointed to the same directory rather than the system library.
